src/storage/mvcc/txn.rs

```diff
@@ -199,14 +178,40 @@ impl<S: Snapshot> MvccTxn<S> {
         Ok(())
     }
 
+    fn handle_non_pessimistic_lock_conflict(
```

Comment:
Please add a comment to explain what this function do and when this function is needed..